site/en/docs/extensions/reference/declarativeNetRequest/index.md

```diff
@@ -221,9 +221,23 @@ manifest field to limit which Chrome versions can install your extension.
 
 The number of rules available after that depends on how many rules are enabled by all the extensions installed on a user's browser. You can find this number at runtime by calling [`getAvailableStaticRuleCount()`](#method-getAvailableStaticRuleCount). You can see [an example of this](#update-static-rulesets) under [code examples](#code-examples).
 
-#### Dynamic and session rules {: #dynamic-session-rules }
+#### Session rules {: #session-rules }
 
-The limits applied to dynamic and session rules are simpler than static rules. The total number of both cannot exceed 5000. This is called the [`MAX_NUMBER_OF_DYNAMIC_AND_SESSION_RULES`](#property-MAX_NUMBER_OF_DYNAMIC_AND_SESSION_RULES).
+An extension can have up to 5000 session rules. This is exposed as the
+[`MAX_NUMBER_OF_SESSION_RULES`](#property-MAX_NUMBER_OF_SESSION_RULES).
+
+Before Chrome 120, there was a limit of 5000 combined dynamic and session rules.
+
+#### Dynamic rules {: #dynamic-rules }
+
+An extension can have at least 5000 dynamic rules. This is exposed as the
+[`MAX_NUMBER_OF_UNSAFE_DYNAMIC_RULES`](#property-MAX_NUMBER_OF_UNSAFE_DYNAMIC_RULES).
+
+Starting in Chrome 121, there is a larger limit of 30,000 rules available for safe dynamic rules,
```

Comment:
Do these include the 5000 unsafe rules?

Reply:
It does. I added a note on that, hopefully it helps.